### PR TITLE
Phase2 hgx186 Define a new scenario D41

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Define a new scenario D41 with the more recent version of HGCal geometry (C8). Also changes are made to overall system (O3 derived from O2) and MTD (I9 derived from I7) which need to be changed

#### PR validation:

Overlap test is completed and with changes in ETL geometry and once the PR 

#### if this PR is a backport please specify the original PR#26153 added there is no overlap in calorimeter region

If this PR is integrated we can close PR #26155

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda please rerun the geometry script to make sure everything is consistent

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by bsunanda** [comment]
@pgunnell, @prebello, @cvuosalo, @ianna,  @franzoni,  @zhenhu Could you please approve this PR

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
@pgunnell, @prebello, @zhenhu ping

**Comment by fabiocos** [comment]
@bsunanda @kpedro88 some comments:

- I will update the mtd.xml file to move the definition of LYSO to https://cmssdt.cern.ch/lxr/source/Geometry/CMSCommonData/data/materials.xml#4679

- I would like to have in Configuration/PyReleaseValidation/python/relval_2023.py a test workflow (corresponding to that for other scenarios, to be usable in routine PR tests and IB);

- as already mentioned, I still see a problem:
```
13:28 cmsdev25 1819> runTheMatrix.py -n --what upgrade 
ignoring non-requested file relval_standard
ignoring non-requested file relval_highstats
ignoring non-requested file relval_pileup
ignoring non-requested file relval_generator
ignoring non-requested file relval_extendedgen
ignoring non-requested file relval_production
ignoring non-requested file relval_ged
processing relval_upgrade
ERROR reading file: relval_upgrade 'DigiFullTriggerPU_2023D41PU'
Traceback (most recent call last):
  File "/build/fabiocos/106X/26212/CMSSW_10_6_X_2019-03-21-1100/bin/slc7_amd64_gcc700/runTheMatrix.py", line 338, in <module>
    ret = runSelected(opt)
  File "/build/fabiocos/106X/26212/CMSSW_10_6_X_2019-03-21-1100/bin/slc7_amd64_gcc700/runTheMatrix.py", line 23, in runSelected
    mrd.prepare(opt.useInput, opt.refRel, opt.fromScratch)
  File "/build/fabiocos/106X/26212/CMSSW_10_6_X_2019-03-21-1100/python/Configuration/PyReleaseValidation/MatrixReader.py", line 493, in prepare
    self.readMatrix(matrixFile, useInput, refRel, fromScratch)
  File "/build/fabiocos/106X/26212/CMSSW_10_6_X_2019-03-21-1100/python/Configuration/PyReleaseValidation/MatrixReader.py", line 237, in readMatrix
    if self.relvalModule.steps[stepName] is None:
KeyError: 'DigiFullTriggerPU_2023D41PU'
```

Its reason is not immediately obvious to me, looking into the code, but I am unable to the use runTheMatrix.py when merging this PR. @bsunanda you said you succeeded to run a test workflow, am I correct?

**Comment by kpedro88** [comment]
@fabiocos I sent a PR to @bsunanda that addresses most of this: https://github.com/bsunanda/cmssw/pull/13
* fixes the bug you observe
* adds workflow to relval_2023
* incorporates new trigger map from @jbsauvan to make the workflow run

Once this is included, this PR should be tested with the cmsdist PR corresponding to https://github.com/cms-data/L1Trigger-L1THGCal/pull/12

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
please test with cms-sw/cmsdist#4799

**Comment by kpedro88** [comment]
abort test

**Comment by kpedro88** [comment]
please test with cms-sw/cmsdist#4799

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2923,6 +2923,7 @@ def gen2018HiMix(fragment,howMuch):
 defaultDataSets['2023D38']=''
 defaultDataSets['2023D39']=''
 defaultDataSets['2023D40']=''
+defaultDataSets['2023D41']=''
```
Comment (fabiocos) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @kpedro88 ok, this was the missing piece causing the failure

**Comment by kpedro88** [comment]
please test workflow 29034.0 with cms-sw/cmsdist#4799

**Comment by bsunanda** [comment]
@cmsbuild please test with cms-sw/cmsdist#4799

**Comment by bsunanda** [comment]
@cmsbuild please test workflow 29034.0 with cms-sw/cmsdist#4799

**Comment by fabiocos** [comment]
+operations

the new scenario is correctly added and usable in the test sequence

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
@cvuosalo I assume this is ok for you
@kpedro88 this looks ok to me to move forward
@rekovic the L1T part looks the straightforward addition of new external, it needs your signature
@prebello @zhenhu the matrix looks now functional

@emiglior do you still see D40 useful?

**Comment by rekovic** [comment]
+1

**Comment by prebello** [comment]
+1

Tests approved and the matrix looks functional. Thank you

**Comment by fabiocos** [comment]
@cvuosalo unless you have objections or further comments, I would like to integrate this PR for 10_6_0_pre3, so as to enable tests

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**